### ISSUE TITLE
Menu: Accessibility rule F85. Lost focus on TAB

### ISSUE THREAD

**Issue by SoyDiego** [primefaces/primeng#14246]
### Describe the bug

`Menu` does not comply with [Accessibility rule F85](https://www.w3.org/WAI/WCAG22/Techniques/failures/F85.html).
On press `TAB` key, should go to the next element.

But if you are using `Menu` component inside a `Table` or another element and in the properties you are using `appendTo="body"`, when you press `TAB` key, loose the focus completely and you cannot continue navigate inside the table.

### (I'm opening the menu with `Spacebar` and then I press `TAB` and loose the focus. Should go to the next row or element.)

![problem accessibility](https://github.com/primefaces/primeng/assets/19764334/544fd826-8e2c-4495-b5cb-2ac7864b004e)

### Environment

-

### Angular version

Last version

### PrimeNG version

Last version

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.16.18

### Browser(s)

All browsers

### Steps to reproduce the behavior

1. Create a Table
2. Create Button inside the table and when you click in the button should add a menu component.
3. If you are moving with the keyboard keys, when you enter to the menu and press tab, you should be in the next row but will loose the focus.

### Expected behavior

When you press `TAB` go to the next element.
